instrumentation/github.com/gocql/gocql/otelgocql/observer.go

```diff
@@ -71,7 +74,7 @@ func (o *OTelQueryObserver) ObserveQuery(ctx context.Context, observedQuery gocq
 
 		ctx, span := o.tracer.Start(
 			ctx,
-			observedQuery.Statement,
+			o.spanNameFormatter(observedQuery),
```

Comment:
I would be a bit more defensive about these not being nil.  If a user does `With*SpanNameFormatter(nil)` this will panic.

I would suggest adding a nil check either after apply's or before used in creating the Observers.